src/devtools/client/debugger/src/components/Editor/LineHitCounts.tsx

```diff
@@ -167,89 +166,106 @@ function LineHitCounts({ sourceEditor }: Props) {
+          let className = styles.HitsBadge0;
+          let index = NUM_GRADIENT_COLORS - 1;
           if (hitCount > 0) {
-            sourceEditor.codeMirror.removeLineClass(lineHandle, "line", "empty-line");
-          } else {
-            // If this line wasn't hit any, dim the line number,
-            // even if it's a line that's technically reachable.
-            sourceEditor.codeMirror.addLineClass(lineHandle, "line", "empty-line");
+            if (minHitCount !== maxHitCount) {
+              index = Math.min(
+                NUM_GRADIENT_COLORS - 1,
+                Math.round(
+                  ((hitCount - minHitCount) / (maxHitCount - minHitCount)) * NUM_GRADIENT_COLORS
+                )
+              );
+            }
+            className = styles[`HitsBadge${index + 1}`];
           }
-        }
 
-        const info = editor.lineInfo(lineNumber);
+          if (features.disableUnHitLines) {
+            if (hitCount > 0) {
+              sourceEditor.codeMirror.removeLineClass(lineHandle, "line", "empty-line");
+            } else {
+              // If this line wasn't hit any, dim the line number,
+              // even if it's a line that's technically reachable.
+              sourceEditor.codeMirror.addLineClass(lineHandle, "line", "empty-line");
+            }
+          }
 
-        let markerNode: HTMLDivElement;
-        if (info?.gutterMarkers?.["hit-markers"]) {
-          // Retrieve the marker DOM node we already created
-          markerNode = info.gutterMarkers["hit-markers"];
-        } else {
-          markerNode = document.createElement("div");
-          markerNode.onclick = () =>
-            updateHitCountsMode(isCollapsedRef.current ? "show-counts" : "hide-counts");
+          const info = editor.lineInfo(lineHandle);
 
-          editor.setGutterMarker(lineHandle, "hit-markers", markerNode);
-        }
+          let markerNode: HTMLDivElement;
+          if (info?.gutterMarkers?.["hit-markers"]) {
+            // Retrieve the marker DOM node we already created
+            markerNode = info.gutterMarkers["hit-markers"];
+          } else {
+            markerNode = document.createElement("div");
+            markerNode.onclick = () =>
+              updateHitCountsMode(isCollapsedRef.current ? "show-counts" : "hide-counts");
 
-        markerNode.className = className;
-        if (!isCollapsed) {
-          let hitsLabel = "";
-          if (hitCount > 0) {
-            hitsLabel = hitCount < 1000 ? `${hitCount}` : `${(hitCount / 1000).toFixed(1)}k`;
+            editor.setGutterMarker(lineHandle, "hit-markers", markerNode);
           }
-          markerNode.textContent = hitsLabel;
-        }
-        markerNode.title = `${hitCount} hits`;
+
+          markerNode.className = className;
+          if (!isCollapsed) {
+            let hitsLabel = "";
+            if (hitCount > 0) {
+              hitsLabel = hitCount < 1000 ? `${hitCount}` : `${(hitCount / 1000).toFixed(1)}k`;
```

Comment:
this feels like it could be a nice helper function given how complex everything around already is, but I don't feel too strongly.